umbral/pre.py

```diff
@@ -206,13 +155,71 @@ def __hash__(self) -> int:
         component_bytes = tuple(component.to_bytes() for component in self.components())
         return hash(component_bytes)
 
-    def __len__(self) -> int:
-        return len(self._attached_cfrags)
-
     def __repr__(self):
         return "{}:{}".format(self.__class__.__name__, hex(int(self.bn_sig))[2:17])
 
 
+class PreparedCapsule:
```

Comment:
I have, it's mentioned in the PR description :) Don't know how justified it is. They are used in different contexts, after all.

Reply:
Why subclass?  Do they actually share any logic?

And if the answer is "no" - and I think it is - then is thing construction even a "Capsule" at all?  And is it even necessary?

Reply:
"Why subclass? Do they actually share any logic?" - I agree, that's what I thought too. Essentially, it's a `CapsuleThatCanVerifyFrags`. Perhaps it will be better to have a `Capsule` and `VerificationMetadata`, with the latter including delegating, receiving, and signing keys?